### PR TITLE
User login sign up

### PR DESCRIPTION
**Pull request by alkhatibmahmoud**
# Description
User log in, user sign up, user auth middleware (not tested). 

image

### Please add ticket link

### CODE REVIEW

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -1,14 +1,54 @@
-// userController.js
```
Comment (alkhatibmahmoud) [inline backend/controllers/userController.js:1]: I just wanted to do checks here for returning correct status codes, let me know if I should move them to the model

File: backend/middleware/auth.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const token =
+        req.body.token || req.query.token || req.headers["x-access-token"];
+
+    if (!token) {
+        return res.status(403).send("A token is required for authentication");
+    }
+    try {
+        //TODO Set up a TOKEN_KEY env variable in docker
+        const decoded = jwt.verify(token, process.env.TOKEN_KEY);
+        req.user = decoded;
```
Comment (serajsas) [inline backend/middleware/auth.js:15]: Does the user have user.id, that identifies the user?

**Review by serajsas — APPROVED** [review]
Left one comment, otherwise LGTM

File: backend/middleware/auth.js (hunk 2 above)
Comment (alkhatibmahmoud) [reply backend/middleware/auth.js:15]: Yes

File: backend/models/userModel.js [hunk 3]
```diff
@@ -1,20 +1,67 @@
...
+    password: { type: String },
+    avatar: { type: String }, // Assuming the avatar is stored as a file path or URL
+    decks: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Deck' }],
 });
 
-const UserModel = mongoose.model('User', UserSchema);
+userSchema.statics.createUser = async function (first_name, last_name, email, password) {
+    try {
+        const newUser = new this({
+            id: uuid(),
```
Comment (alkhatibmahmoud) [inline backend/models/userModel.js:24]: @serajsas

**Pull request merged**